contracts/strategies/morpho/MetaMorphoStrategy.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.10;
+
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "morpho-urd/src/interfaces/IUniversalRewardsDistributor.sol";
+import "../../interfaces/IERC20Detailed.sol";
+import "../../interfaces/morpho/IMetamorphoSnippets.sol";
+import "../ERC4626Strategy.sol";
+
+contract MetaMorphoStrategy is ERC4626Strategy {
+  using SafeERC20Upgradeable for IERC20Detailed;
+
+  //// @notice MORPHO governance token
+  address internal constant MORPHO = 0x9994E35Db50125E0DF82e4c2dde62496CE330999;
+
+  /// @notice reward token address (e.g. MORPHO, COMP, ...)
+  address[] public rewardTokens;
+
+  /// @notice MORPHO transferability
+  bool public morphoTransferable;
+
+  /// @notice Metamorpho snippets contract, used for apr
+  address public mmSnippets;
+
+  /// Initialization
+
+  /// @notice initialize the strategy
+  /// @param _strategyToken strategy token address
+  /// @param _token underlying token address
+  /// @param _owner owner address
+  /// @param _mmSnippets snippets contract address
+  /// @param _rewardTokens array of reward tokens
+  function initialize(
+    address _strategyToken,
+    address _token,
+    address _owner,
+    address _mmSnippets,
+    address[] memory _rewardTokens
+  ) public {
+    _initialize(_strategyToken, _token, _owner);
+
+    // used to fetch apr
+    mmSnippets = _mmSnippets;
+
+    for (uint256 i = 0; i < _rewardTokens.length; i++) {
+      rewardTokens.push(_rewardTokens[i]);
+    }
+  }
+
+  /// External and public methods
+
+  /// @notice redeem the rewards
+  /// @return rewards amount of reward that is deposited to the ` strategy`
+  function redeemRewards(bytes calldata data)
+    public
+    virtual
+    override
+    onlyIdleCDO
+    nonReentrant
+    returns (uint256[] memory rewards)
+  {
+    address[] memory _rewardTokens = rewardTokens;
+    uint256 _rewardsLen = _rewardTokens.length;
+    if (_rewardsLen == 0) {
+      return rewards;
+    }
+
+    rewards = new uint256[](_rewardsLen);
+    bytes[] memory claimDatas = abi.decode(data, (bytes[]));
+    address reward;
+    address rewardDistributor;
+    uint256 claimable; 
+    bytes32[] memory proof;
+
+    for (uint256 i = 0; i < _rewardsLen; i++) {
+      if (claimDatas[i].length == 0) {
+        continue;
+      }
+      (reward, rewardDistributor, claimable, proof) = abi.decode(claimDatas[i], (address, address, uint256, bytes32[]));
+      rewards[i] = _claimReward(IUniversalRewardsDistributor(rewardDistributor), reward, claimable, proof);
+    }
+  }
+
+  /// @notice transfer rewards to IdleCDO, used if someone claims on behalf of the strategy
+  /// or to transfer MORPHO once it's transferable
+  /// @dev Anyone can call this function
+  function transferRewards() public {
+    address[] memory _rewardTokens = rewardTokens;
```

Comment:
I don't think this step saves any gas, because you read each item and the length once and once only

But if you copy it to memory, you actually expand memory and write to memory, which costs additional gas compared to storing the item value on the stack inside the loop (and the array length outside of the loop)